packages/go-gen/src/utils.rs

```diff
@@ -0,0 +1,53 @@
+/// An iterator that returns all suffixes of a string, excluding the empty string.
+///
+/// It starts with the full string and ends with the last character.
+/// It is a double-ended iterator and can be reversed.
+pub fn suffixes(s: &str) -> impl Iterator<Item = &str> + DoubleEndedIterator {
+    s.char_indices().map(|(pos, _)| &s[pos..])
+}
+
+/// Replaces common pascal-case acronyms with their uppercase counterparts.
+pub fn replace_acronyms(ty: &str) -> String {
+    let mut ty = ty.replace("Url", "URL");
+    replace_in_place(&mut ty, "Uri", "URI");
+    replace_in_place(&mut ty, "Id", "ID");
+    replace_in_place(&mut ty, "Ibc", "IBC");
+    ty
+}
```

Comment:
Very good. Now I assume `replace_acronyms("MyIdentity")` would not return the expected result since the name is not split in words first